python/dpu_utils/tf2utils/mlp.py

```diff
@@ -30,7 +31,16 @@ def __init__(
         """
         super().__init__()
         if isinstance(hidden_layers, int):
-            self._hidden_layer_sizes = [out_size] * hidden_layers
+            if out_size == 1:
+                print(
+                    f"W: In {name}, was asked to use {hidden_layers} layers of size 1, which is most likely wrong."
+                    f" Switching to {hidden_layers} layers of size 32; to get hidden layers of size 1,"
+                    f" use hidden_layers=[1,...,1] explicitly.",
+                    file=sys.stderr,
+                )
+                self._hidden_layer_sizes = [32] * hidden_layers
```

Comment:
this looks good, but
a) why not `assert out_size != 1` instead
b) maybe use `logging.warn()` instead of print?

Reply:
Re (a): I dislike adding an assertion for something post-hoc, as it may break something; (b) we are not using logging cleanly anyway (cf. uses in RichPath, for example).
So, you're right in principle, but I want to keep things in line with what's here already :-)